src/main/java/com/github/kiulian/downloader/model/search/SearchResultItem.java

```diff
@@ -0,0 +1,54 @@
+package com.github.kiulian.downloader.model.search;
+
+public interface SearchResultItem {
+
+    String title();
+
+    default boolean isVideo() {
```

Comment:
Please, explain why do we need 4 boolean methods instead of single `type()`?

Reply:
Good call. A `SearchResultItemType` enum would do the job and be handy for parsing.
I like the pattern of boolean methods overridden in subclasses, but I'll remove them if you want.

Reply:
They're gone, only one `type()` method remains.